src/contract.rs

```diff
@@ -86,55 +100,68 @@ pub fn try_add(
     .unwrap_or(false);
 
     if !valid {
-        return Err(HandleError::InvalidSignature {});
+        return Err(ContractError::InvalidSignature {});
     }
 
-    let randomness = derive_randomness(&signature);
+    let randomness: [u8; 32] = derive_randomness(&signature);
```

Comment:
This should not be needed because derive_randomness always returns `[u8; 32]`

```suggestion
    let randomness = derive_randomness(&signature);
```

Reply:
Fixed